### PR TITLE
Use hugo pipes for asset processing

### PR DESCRIPTION
**Pull request by davidkroell**
### Advantage

This is mainly for speed purpose. By using Hugo pipes, different actions can be applied to the asset files, such as minification of CSS and JS, which should result is better loading time, which in turn increases the usability.

Furthermore, assets which were not found are leading to build errors, not runtime errors. In extend, no more handmade URL-building is required. Also integrity attributes for javascript files are added.

Should also increase developer experience because files in the static folder are sometimes not rebuilt during development with `hugo serve` (at least I experienced it this way).

#### Changes

* move assets from static to assets folder
* add minification for CSS and JS files
* bundle all JS files and generate fingerprint

### CODE REVIEW

File: layouts/partials/head.html [hunk 1]
```diff
@@ -37,13 +37,23 @@
...
+    <link rel="stylesheet" type="text/css" href="{{ $cssNormalize.RelPermalink }}" media="print" onload="this.media='all'">
 
-    <script src="{{ .Site.BaseURL }}/js/svg-injector.min.js"></script>
-    <script src="{{ .Site.BaseURL }}/js/feather-icons.min.js"></script>
-    <script src="{{ .Site.BaseURL }}/js/main.js"></script>
+    {{ $cssMain := resources.Get "css/main.css" | minify }}
+    <link rel="stylesheet" type="text/css" href="{{ $cssMain.RelPermalink }}" onload="this.media='all'">
+
+    {{ $cssDark := resources.Get "css/dark.css" | minify }}
+    <link id="dark-theme" rel="stylesheet" href="{{ $cssDark.RelPermalink }}" onload="this.media='all'">
```
Comment (526avijitgupta) [inline layouts/partials/head.html:47]: Let's not load any CSS files asynchronously. We can get rid of `onload="this.media='all'`. Running into FOUC or visual glitches is far worse than saving a few miliseconds IMO.

**Review by 526avijitgupta — COMMENTED** [review]
one small comment, looks good from my side.

cc @yashmehrotra please review

File: layouts/partials/head.html (hunk 1 above)
Comment (davidkroell) [reply layouts/partials/head.html]: Removed it

**Comment by davidkroell** [comment]
*Edit:* Added bundling of JavaScript files

**Review by 526avijitgupta — APPROVED** [review]
LGTM thanks @davidkroell 

@yashmehrotra Please review

**Comment by yashmehrotra** [comment]
@526avijitgupta @davidkroell

Will this break existing sites since we are changing the base asset path ? For people who added extra assets on their own, will serve from the old static directory

Should we create/pin an issue to handle those queries ?

**Review by yashmehrotra — APPROVED** [review]

**Comment by davidkroell** [comment]
No, does not break anything, since these additional static assets are still served from the same URL and can be embedded withing `customHeadHTML` as usual.

**Pull request merged**